cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -23,41 +23,57 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
-	"log"
+	"net/url"
 	"os"
 	"time"
 
 	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
 	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
 	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
 	kclientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
-	kfields "github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
-	klabels "github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	kcontrollerFramework "github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	kSelector "github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
 	tools "github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
-	kwatch "github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 	etcd "github.com/coreos/go-etcd/etcd"
+	"github.com/golang/glog"
 	skymsg "github.com/skynetservices/skydns/msg"
 )
 
 var (
-	domain                = flag.String("domain", "kubernetes.local", "domain under which to create names")
-	etcd_mutation_timeout = flag.Duration("etcd_mutation_timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
-	etcd_server           = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
-	verbose               = flag.Bool("verbose", false, "log extra information")
-	kubecfg_file          = flag.String("kubecfg_file", "", "Location of kubecfg file for access to kubernetes service")
+	argDomain              = flag.String("domain", "kubernetes.local", "domain under which to create names")
+	argEtcdMutationTimeout = flag.Duration("etcd_mutation_timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
+	argEtcdServer          = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
+	argKubecfgFile         = flag.String("kubecfg_file", "", "Location of kubecfg file for access to kubernetes service")
+	argKubeMasterUrl       = flag.String("kube_master_url", "http://${KUBERNETES_RO_SERVICE_HOST}:${KUBERNETES_RO_SERVICE_PORT}", "Url to reach kubernetes master.")
```

Comment:
Updated to use the regular service.

On Thu, May 7, 2015 at 1:57 PM, Daniel Smith <notifications@github.com>
wrote:

> In cluster/addons/dns/kube2sky/kube2sky.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7901#discussion_r29895168>
> :
>
> >  	skymsg "github.com/skynetservices/skydns/msg"
> >  )
> >
> >  var (
> > -	domain                = flag.String("domain", "kubernetes.local", "domain under which to create names")
> > -	etcd_mutation_timeout = flag.Duration("etcd_mutation_timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
> > -	etcd_server           = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
> > -	verbose               = flag.Bool("verbose", false, "log extra information")
> > -	kubecfg_file          = flag.String("kubecfg_file", "", "Location of kubecfg file for access to kubernetes service")
> > +	argDomain              = flag.String("domain", "kubernetes.local", "domain under which to create names")
> > +	argEtcdMutationTimeout = flag.Duration("etcd_mutation_timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
> > +	argEtcdServer          = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
> > +	argKubecfgFile         = flag.String("kubecfg_file", "", "Location of kubecfg file for access to kubernetes service")
> > +	argKubeMasterUrl       = flag.String("kube_master_url", "http://${KUBERNETES_RO_SERVICE_HOST}:${KUBERNETES_RO_SERVICE_PORT}", "Url to reach kubernetes master.")
>
> RO service is going away.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7901/files#r29895168>
> .
>